faiss/utils/hamming_distance/avx512-inl.h

```diff
@@ -0,0 +1,485 @@
+/*
+ * Copyright (c) Meta Platforms, Inc. and affiliates.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+#ifndef HAMMING_AVX512_INL_H
+#define HAMMING_AVX512_INL_H
+
+// AVX512 version
```

Comment:
Can you add a comment here summarizing the discussions of the two PRs:
```
The _mm512_popcnt_epi64 intrinsic is used to accelerate Hamming distance calculations in HammingComputerDefault and HammingComputer64. This is not available in default FAISS avx512 built mode and is only available on Intel Sapphire Rapids
```

Reply:
Added.